.github/workflows/tests.yaml

```diff
@@ -227,12 +248,15 @@ jobs:
 
       # We need to replace the `.` with a `_` for the build cache.
       - name: Setup LLVM and Clang (macOS)
-        if: steps.filter.outputs.has_code == 'true' && matrix.os == 'macos-12'
+        if:
+          steps.filter.outputs.has_code == 'true' && contains(matrix.os,
+          'macos')
```

Comment:
Why are you using macos-12 for the macos-14 cache? Will the build outputs all be the same, even with the environment and CPU change?

Reply:
No, good catch. I had missed that the version was included here. See below for a full answer.